### ISSUE TITLE
[Feat]: Implementing Conservative Force Field in Graphormer

### ISSUE THREAD

**Issue by panmianzhi** [microsoft/mattersim#99]
### Contact Details

panmianzhi@163.com

### Feature Description

I’ve noticed in MatterSim that the M3GNet model is trained using a conservative force field, whereas Graphormer is trained with a direct force prediction approach. I have a theoretical question regarding Graphormer: is it possible to implement it with a conservative force field?

From my understanding, unlike M3GNet, which calculates edge vectors based on atomic positions, Graphormer uses atomic positions to compute spatial encoding as an attention bias. Given this mechanism, can the energy output of Graphormer be differentiated with respect to atomic positions to derive a conservative force field? I’d appreciate any insights or clarifications on this matter.

Thank you!

### Motivation

The Conservative force field is important for MD. Recent studies propose pretraining MLIP with direct force prediction and fine-tuning using conservative force field. I just wonder Graphformer can be used for conservative FF.

### Contribution Interest

- [x] I'm interested in potentially implementing this feature
- [ ] I can provide guidance, but cannot implement
- [x] I'm just suggesting the feature for the community

### Code of Conduct

- [x] I agree to follow the project's Code of Conduct

**Comment by yanghan234** [comment]
Thank you for your interest in MatterSim. You are correct; the current public version is based on M3GNet, which uses energy gradients to predict forces and ensure energy conservation. 
Unfortunately, I can't share many details about the internal version at this time. Please stay tuned for future updates in this repo!
